pandas/tests/series/test_reductions.py

```diff
@@ -51,6 +51,14 @@ def test_mode_nullable_dtype(any_numeric_ea_dtype):
     tm.assert_series_equal(result, expected)
 
 
+def test_mode_nullable_dtype_edge_case(any_numeric_ea_dtype):
+    # GH##58926
+    ser = Series([1, 1, 2, 3], dtype=any_numeric_ea_dtype)
```

Comment:
Could you add pd.NA to this Series? as I mentioned above this test will fail if you do so. (incorrect masking)

Your initial fix is good, but more investigation needs to be done in `algorithims.mode` which uses `_libs`'s `hashtable.mode` which seems to have the wrong behavior there. I think that's written in C so I couldn't help debug it further

Reply:
Right, thanks for the spot, masking issue seems to be that the original code would just always set one bit of the mask if there's a na value even if it wasn't actually part of the mode.